shuup/campaigns/models/basket_line_effects.py

```diff
@@ -103,7 +107,7 @@ class DiscountFromProduct(BasketLineEffect):
     def description(self):
         return _("Select discount amount and products.")
 
-    def get_discount_lines(self, order_source, original_lines):
+    def get_discount_lines(self, order_source, original_lines, supplier):
         product_ids = self.products.values_list("pk", flat=True)
         campaign = self.campaign
         campaign_supplier = campaign.supplier if hasattr(campaign, "supplier") and campaign.supplier else None
```

Comment:
maybe this `campaign_supplier` should be the parameter and then fallback to the campaign supplier. The same with other places.